--- makefile.orig	2012-08-24 12:13:46.000000000 +0400
+++ makefile	2012-08-24 12:14:35.000000000 +0400
@@ -5,10 +5,11 @@
 #   to create an suid root binary.
 #
 
-APXS = /usr/sbin/apxs
+APXS = /usr/bin/apxs
 CC = gcc
-INSTBINDIR = /usr/local/sbin
-DEFINES = -D INSTBINDIR=\\\"$(INSTBINDIR)\\\"
+CFLAGS = $(RPM_OPT_FLAGS)
+INSTBINDIR = /usr/sbin
+DEFINES = -DINSTBINDIR=\\\"$(INSTBINDIR)\\\"
 
 all: validate .libs/mod_auth_shadow.so
 
@@ -22,18 +23,16 @@
 test: testvalidate
 
 validate: validate.o
-	$(CC) validate.o -lcrypt -o validate
-	chown root:root validate
-	chmod +s validate
+	$(CC) $(CFLAGS) validate.o -lcrypt -o validate
 
 validate.o: validate.c validate.h
-	$(CC) -c validate.c
+	$(CC) $(CFLAGS) -c validate.c
 
 testvalidate.o: testvalidate.c
-	$(CC) -c testvalidate.c
+	$(CC) $(CFLAGS) -c testvalidate.c
 
 testvalidate: testvalidate.o
-	$(CC) testvalidate.o -o testvalidate
+	$(CC) $(CFLAGS) testvalidate.o -o testvalidate
 
 .libs/mod_auth_shadow.so: mod_auth_shadow.c validate.h
 	$(APXS) $(DEFINES) -c mod_auth_shadow.c
